aten/src/THC/generic/THCTensorMath.cu

```diff
@@ -461,6 +461,8 @@ void THCTensor_(range)(THCState *state, THCTensor *r_, accreal xmin, accreal xma
 void THCTensor_(arange)(THCState* state, THCTensor *r_, accreal xmin, accreal xmax, accreal step) {
   THCAssertSameGPU(THCTensor_(checkGPU)(state, 1, r_));
   THArgCheck(step > 0 || step < 0, 3, "step must be nonzero");
+  THArgCheck(std::isfinite(xmin) && std::isfinite(xmax)
+              , 1, "unsupported range: ", xmin, " -> ", xmax);
   THArgCheck(((step > 0) && (xmax >= xmin)) || ((step < 0) && (xmax <= xmin))
               , 2, "upper bound and larger bound inconsistent with step sign");
   ptrdiff_t size = (ptrdiff_t) ceil(ScalarConvert<accreal, double>::to(xmax - xmin) / step);
```

Comment:
Can this still overflow `ptrdiff_t` if we pass something like torch.arange(1.175494351e-38, 3.402823466e+38)? Can `xmax-xmin` itself turn to `inf` if `xmin` is negative and `xmax - xmin` does not fit in double?

Reply:
Even with good finite non-infs:
```
torch.arange(1.175494351e-38, 3.402823466e+38).shape
torch.Size([-9223372036854775808])
```

Reply:
hmm... it seems like overflow is supposed to be caught by THGeneral.cpp:191:

```
>>> x = torch.arange(0, 9.223372036854775E18)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: $ Torch: invalid memory size -- maybe an overflow? at /home/bvaughan/repos/pytorch/aten/src/TH/THGeneral.cpp:191

```